### PR TITLE
Upgrading to Holocron v1.1.0 

### PR DESCRIPTION
**Pull request by mtomcal**
A test of the Holocron v1.1.0 release from https://github.com/americanexpress/holocron/pull/21 that fixes https://github.com/americanexpress/one-app/issues/57

## Description

The Holocron v1.1.0 has been tested locally with integration and unit tests to ensure no breaking changes. The routing and prefetching integration tests have been modified to test if `loadModuleData` is indeed ran on client side module routing and prefetching inside `healthy-frank` and `ssr-frank`.

I have modified one file that relies on `MODULE_LOADED` constant from Holocron's private API which has been moved from `holocron/ducks/load` to `holocron/ducks/constants` file. 

## Motivation and Context
This fixes issue documented in https://github.com/americanexpress/one-app/issues/57 related to data loading on client side loading of Modules.

## How Has This Been Tested?
Integration Tests and Unit Tests.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

The `loadModuleData` function will be called not only on the serverside load but clientside load now as well. Holocron has introduced the new Holocron Configuration API as well. We drop bundle size by ~1 Kb ungzipped as we no longer need to import `holocronModule` from Holocron inside Modules but instead rely on a static to configure Holocron. (See https://github.com/americanexpress/holocron/pull/21).

TODOs:
- [ ] Release Holocron v1.1.0 and update this PR so tests can pass in CI/CD
- [x] Update API Documentation in Loading Data

### CODE REVIEW

**Comment by mtomcal** [comment]
The tests will not pass until a release from Holocron happens. It has been tested locally to pass unit and integration tests from a local build of the future Holocron v1.1.0.

File: __tests__/integration/one-app.spec.js [hunk 1]
```diff
@@ -799,30 +799,29 @@ describe('Tests that can run against either local Docker setup or remote One App
           expect(url).toMatch(/healthy-frank\/simple/);
         });
 
-        test('loadModuleData is not called when navigating to module on client', async () => {
+        test('data loads when navigating to module on client', async () => {
           // start by navigating to ssr-frank without prefetch
           await browser.url(`${appInstanceUrls.browserUrl}/healthy-frank`);
-          const noPrefetchLink = await browser.$('.ssr-frank-link');
-          await noPrefetchLink.click();
+          const prefetchLink = await browser.$('.prefetch-ssr-frank');
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]: Ideally this should be testing navigating without potential prefetching.

Comment (mtomcal) [reply __tests__/integration/one-app.spec.js]: Fixed in f1b0889 by modifying test to be more specific.

**Comment by 10xLaCroixDrinker** [comment]
LGTM pending passing build

**Comment by mtomcal** [comment]
Updated to Holocron v1.1.0 and documentation has been updated to account for API changes. Waiting on slow Travis to build but once done, should be good for review.

**Comment by infoxicator** [comment]
I found a potential bug when developing with the new Holocron config API:
1. Make a change to SSR frank
2. `npm run build`
3. Browser reload the route `http://localhost:3000/healthy-frank/ssr-frank`

**Comment by 10xLaCroixDrinker** [comment]
@mtomcal, the issue @infoxicator found will block this PR

**Comment by 10xLaCroixDrinker** [comment]
I pushed a commit that fixes the issue @infoxicator found. @infoxicator to confirm before merging

**Comment by infoxicator** [comment]
> I pushed a commit that fixes the issue @infoxicator found. @infoxicator to confirm before merging

Tested and working as expected during local development

File: docs/recipes/Code-Splitting-Using-Holocron.md [hunk 2]
```diff
@@ -189,40 +189,35 @@ root
...
+      // the same dynamic import is used to create a promise with our chunk
+      const imported = await import(/* webpackChunkName: '<chunkName>' */ './Chunk');
+      Chunk = imported.default || imported;
+    } catch (error) {
+      Chunk = () => <p>{error.message}</p>;
+    }
+  },
   // we can wait for our holocron module to load the chunk during server-side rendering
   // and render the module with the loaded chunk
-  options: { ssr: true },
```
Comment (JAdshead) [inline docs/recipes/Code-Splitting-Using-Holocron.md:224]: this was only relevant to the `load` function right ? is there a case to disable calling `loadModuleData` on the server ?

File: prod-sample/sample-modules/ssr-frank/0.0.0/src/components/SsrFrank.jsx [hunk 3]
```diff
@@ -26,7 +25,12 @@ const SsrFrank = ({ moduleState }) => (
   </div>
 );
 
-SsrFrank.loadModuleData = async ({ store, fetchClient }) => {
+const loadModuleData = async ({ store, fetchClient }) => {
+  const moduleState = store.getState().getIn(['modules', 'ssr-frank']);
+  // If isComplete and data already exists dont run request again
+  if (moduleState.get('isComplete') && moduleState.get('data')) {
```
Comment (JAdshead) [inline prod-sample/sample-modules/ssr-frank/0.0.0/src/components/SsrFrank.jsx:31]: is this necessary ?

File: docs/recipes/Code-Splitting-Using-Holocron.md (hunk 2 above)
Comment (mtomcal) [reply docs/recipes/Code-Splitting-Using-Holocron.md:224]: Yes those were only for `load`. 

And Yes:
```js
MyModule.holocron = {
  loadModuleData: global.BROWSER ? loadModuleData : undefined;
}
```

File: prod-sample/sample-modules/picky-frank/0.0.1/package.json [hunk 4]
```diff
@@ -9,6 +9,7 @@
     "prepare": "npm run build"
   },
   "dependencies": {
+    "holocron": "^1.1.0",
```
[the remark this answers is not in the document]
Comment (mtomcal) [reply prod-sample/sample-modules/picky-frank/0.0.1/package.json:12]: Dont need it as a dep anymore in Modules when using Holocron config static.

File: prod-sample/sample-modules/ssr-frank/0.0.0/src/components/SsrFrank.jsx (hunk 3 above)
Comment (mtomcal) [reply prod-sample/sample-modules/ssr-frank/0.0.0/src/components/SsrFrank.jsx:31]: Do you wish for this to double load on both server and client?

Comment (JAdshead) [reply prod-sample/sample-modules/ssr-frank/0.0.0/src/components/SsrFrank.jsx:31]: true true

**Pull request merged**